ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintConfigurationProcessor.java

```diff
@@ -359,152 +362,245 @@ private boolean containsHostFromHostGroups(String configType, String propertyNam
     // removes a property other than the property it is registered for then we will
     // have an issue as it won't be removed from the clusterProps map as it is a copy.
     Map<String, Map<String, String>> clusterProps = clusterConfig.getFullProperties();
-    for (Map<String, Map<String, PropertyUpdater>> updaterMap : createCollectionOfUpdaters()) {
+    doGeneralPropertyUpdatesForClusterCreate(clusterConfig, clusterProps, configTypesUpdated);
+
+    //todo: lots of hard coded HA rules included here
+    if (clusterTopology.isNameNodeHAEnabled()) {
+      doNameNodeHAUpdateOnClusterCreation(clusterConfig, clusterProps, configTypesUpdated);
+    }
+
+    // Explicitly set any properties that are required but not currently provided in the stack definition.
+    setStackToolsAndFeatures(clusterConfig, configTypesUpdated);
+    setRetryConfiguration(clusterConfig, configTypesUpdated);
+    setupHDFSProxyUsers(clusterConfig, configTypesUpdated);
+    addExcludedConfigProperties(clusterConfig, configTypesUpdated, clusterTopology.getBlueprint().getStack());
+
+    trimProperties(clusterConfig, clusterTopology);
+
+    return configTypesUpdated;
+  }
+
+  /**
+   * Update Namenode HA properties during cluster creation
+   * @param clusterConfig
+   * @param clusterProps
+   * @param configTypesUpdated
+   * @throws ConfigurationTopologyException
```

Comment:
Please don't add such javadoc tags without description (as it doesn't add any value to the information already available from the signature).  Please either add description, or drop the tag.